From d82a5b4e2efade13079b00205d77cc7360737c18 Mon Sep 17 00:00:00 2001
From: Tony Cheng <tony.cheng@amd.com>
Date: Thu, 25 Aug 2016 21:27:56 -0400
Subject: [PATCH 0939/1722] drm/amd/dal: remove obsolete vbios code

Remove headless_no_opm.  this appears to be obsolete code supporting dual ASIC
crossfire with DVO (CF connector).  DVO CF is not supported in DAL3

Change-Id: Iaad24a53ddb84655aa7e1a090d8f145e21b68ef9
Signed-off-by: Jordan Lazare <Jordan.Lazare@amd.com>
Acked-by: Harry Wentland <harry.wentland@amd.com>
Signed-off-by: Kalyan Alle <kalyan.alle@amd.com>
---
 drivers/gpu/drm/amd/dal/dc/bios/bios_parser.c                | 12 ------------
 drivers/gpu/drm/amd/dal/dc/bios/bios_parser_types_internal.h |  1 -
 2 files changed, 13 deletions(-)

diff --git a/drivers/gpu/drm/amd/dal/dc/bios/bios_parser.c b/drivers/gpu/drm/amd/dal/dc/bios/bios_parser.c
index 5554c54..bb78d3c 100644
--- a/drivers/gpu/drm/amd/dal/dc/bios/bios_parser.c
+++ b/drivers/gpu/drm/amd/dal/dc/bios/bios_parser.c
@@ -3605,18 +3605,6 @@ static enum bp_result patch_bios_image_from_ext_display_connection_info(
 	if (get_ext_display_connection_info(bp, as,
 					    opm_object,
 					    &ext_display_connection_info_tbl) != BP_RESULT_OK) {
-		if (bp->headless_no_opm) {
-			/* Failed to read OPM, remove all non-CF connectors. */
-			for (i = 0; i < connector_tbl->ucNumberOfObjects; ++i) {
-				object = &connector_tbl->asObjects[i];
-				object_id = object_id_from_bios_object_id(
-						le16_to_cpu(object->usObjectID));
-				if (OBJECT_TYPE_CONNECTOR == object_id.type)
-					object->usObjectID = cpu_to_le16(0);
-			}
-
-			return BP_RESULT_OK;
-		}
 
 		dal_logger_write(bp->base.ctx->logger,
 				LOG_MAJOR_BIOS,
diff --git a/drivers/gpu/drm/amd/dal/dc/bios/bios_parser_types_internal.h b/drivers/gpu/drm/amd/dal/dc/bios/bios_parser_types_internal.h
index 69e11b2..be12ac7 100644
--- a/drivers/gpu/drm/amd/dal/dc/bios/bios_parser_types_internal.h
+++ b/drivers/gpu/drm/amd/dal/dc/bios/bios_parser_types_internal.h
@@ -67,7 +67,6 @@ struct bios_parser {
 	uint8_t *bios_local_image;
 
 	bool remap_device_tags;
-	bool headless_no_opm;
 };
 
 /* Bios Parser from DC Bios */
-- 
2.7.4

